src/main/java/pl/javastart/task/firma/Firma.java

```diff
@@ -0,0 +1,84 @@
+package pl.javastart.task.firma;
+
+import pl.javastart.task.opodatkowanie.*;
+
+import java.util.Arrays;
+
+public class Firma {
+
+    private static final int ILOSC_ELEMENTOW_TABLICY = 10;
+
+    private final String nazwa;
+    private final FormaOpodatkowania formaOpodatkowania;
+
+    private OperacjaFinansowa[] przychody = new OperacjaFinansowa[ILOSC_ELEMENTOW_TABLICY];
+    private OperacjaFinansowa[] wydatki = new OperacjaFinansowa[ILOSC_ELEMENTOW_TABLICY];
+    private int aktualnaIloscPrzychodow = 0;
+    private int aktualnaIloscWydatkow = 0;
+
+    public Firma(String nazwa, FormaOpodatkowania formaOpodatkowania) {
+        this.nazwa = nazwa;
+        this.formaOpodatkowania = formaOpodatkowania;
+    }
+
+    public void wyswietlPodsumowanie() {
+        double sumaPrzychodow = zsumujPrzychody();
+        double sumaWydatkow = zsumujWydatki();
+
+        System.out.printf("======= Firma: %s ===========\n", nazwa);
+        System.out.printf("Forma opodatkowania: %s\n", formaOpodatkowania.getTaxForm());
+        System.out.printf("Suma przychodów: %.2f zł\n", sumaPrzychodow);
+        System.out.printf("Suma wydatków: %.2f zł\n", sumaWydatkow);
+
+        double kwotaBazowa;
+
+        if (formaOpodatkowania instanceof NaCzarno || formaOpodatkowania instanceof RyczaltEwidencjonowany) {
+            kwotaBazowa = sumaPrzychodow;
+        } else {
+            kwotaBazowa = sumaPrzychodow - sumaWydatkow;
+        }
```

Comment:
Lepiej zmienić sygnaturę metody wyliczPodatek w taki sposób, żeby przyjmowała dodatkowo wydatki, niż używać tutaj instanceof

Reply:
Faktycznie to będzie dobry pomysł na ominięcie użycia instanceOf.